src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/util/VisibleMemberTable.java

```diff
@@ -1012,7 +1012,6 @@ public ImplementedMethods(ExecutableElement method) {
                 //  this needs to be fixed
```

Comment:
I can't add this comment where I'd like to ... line 992, the declaration of the `ImplementedMethods` class.
Especially given the "curious" nature of the constructor, it would help to have descriptive comments on either the class or constructor.  It seems like `ImplementedMethods` should be a collection of methods implemented by a `TypeElement` (similar to filtering `Element.getEnclosedElements`. But is seems like this collection is more like the collection of methods in a class hierarchy that :"match" a given signature .. perhaps too get a candidate list of methods rom which to gather documentation (e.g with `{@inheritDoc}`).

Given that, would it be a reasonable alternative rationale for removing the `removeOverriddenMethod` because by construction, nothing ever gets added that subsequently needs to be removed?

I'm still trying to understand this one. I think I know what it is trying to do, but I'm concerned that while it might be trying to do something useful, it might be implemented incorrectly ... in which case, it should be fixed, not removed.

I guess I'd be interest to hear the actual/intended operation when a collection of different possibly-unrelated interfaces all implement the "same" method according to its signature.   Think: the "graphical cowboy" issue.   Two unrelated interfaces that both implement the `draw()` method. Now create a hierarchy of subtypes of those interfaces that start overriding methods with either simple overrides on not-simple overrides.  Then see if `removeOverriddenMethod` kicks in. My guess is that this situation is uncommon and does not arise in JDK API or even our tests.

Reply:
> I'm not convinced by this one. I'd like to see more investigation.

I couldn't find a case where `removeOverriddenMethod` removed an element from `methlist`. I tried to find it by reasoning and by debugging our tests, which in conjunction convinced me that `methlist.remove(i)` is currently unreachable.

Has it ever been reachable? This might depend on how `Utils.overriddenClass` and its predecessors were implemented. Had they ever been capable of returning an *interface* rather than only a class? If not, I cannot see how this could ever work.

If at some point `overriddenClass` could return an interface, then my hypothesis is that those two methods, removeOverriddenMethod and overridingMethodFound, ensured that we constructed `methlist` in such a way that it only contained the *most derived* methods. That is, that for every method m1 from `methlist` there were no method m2 in `methlist` such that m2 overrode m1.

I think that at this point `ImplementedMethods` can work with these two methods being broken because of the way override-candidates are collected by `utils.getAllInterfaces`.